### DISCUSSION TITLE
Increase max vertex resolution in terrain

### DISCUSSION THREAD

**Discussion by Dgzt** [JamesTKhan/Mundus#177] [General]
The max terrain vertex resolution that the Mundus can handle is 256. This is [limited by libgdx](https://github.com/libgdx/libgdx/pull/5925). I think we can split vertices into multiple meshpart in [init method in Terrain file](https://github.com/JamesTKhan/Mundus/blob/8b7d724a62055789a3cdd11b61e8e71ce0d72977/commons/src/main/com/mbrlabs/mundus/commons/terrain/Terrain.java#L110). But I don't know how can split material to meshparts. What do you think, is it possible?

**Comment by Dgzt** [comment]
Seems the multiple meshparts doesn't work, because use [mesh's vertices and indices](https://javadoc.io/doc/com.badlogicgames.gdx/gdx/latest/com/badlogic/gdx/graphics/g3d/model/MeshPart.html).

**Comment by JamesTKhan** [comment]
You would want them to be separate terrain tiles/chunks so that they can be culled for performance reasons. 

I've experimented with adding a "terrain system" in the past to mundus that uses a chunking system but the UI and editing gets complicated maybe that can be revisited I shelved the code 

 ![unknown-7.png](https://github.com/JamesTKhan/Mundus/assets/28971753/881e0f1f-b57b-4145-9d00-b2c1b1e545c0)

![unknown-8.png](https://github.com/JamesTKhan/Mundus/assets/28971753/3fc8d3bb-ad23-4bf2-8810-49abe4515d3d)

Maybe I can create a draft PR and get your thoughts and maybe help finish it if interested

**Comment by Dgzt** [comment]
Yes, this can be a good solution for big maps. If you create a draft PR I will help to you finish with it. Seems the usual map ( indices <= 256) is too small for me, so If I want to use Mundus in my game(s) then I need a solution like this.

**Reply by JamesTKhan** [reply]
Alright, it might be a little while until I can publish something for this but its on my list of things to do

**Comment by JamesTKhan** [comment]
@Dgzt I retrieved some of my code on the terrain chunking from last year, I tried to clean it up a little bit and I got it caught up with master branch. It is still very rough around the edges. Here is a video of what it does currently. I pushed to terrain_chunks branch (https://github.com/JamesTKhan/Mundus/tree/terrain_chunks)

If you are interested in working on it let me know. Can chat about it, talk about ideas and what it still needs.

Right now what I've done is:
1. Add FastNoiseLite library for noise gen (more options than plan old Perlin)
2. The Noise Widget has a convenient Noise Preview to show how a single terrain tile will look.
3. A "modifier" system of sorts, that allows you to layer multiple NoiseModifiers so you can have layers of noise on generation. Not fully worked out yet though.
4. When terrains are created, multiple threads are used to prevent the application from locking up. A Parent GO is created first, and all the child terrain chunks are added under that parent for organization.

https://github.com/JamesTKhan/Mundus/assets/28971753/ad20862a-3a79-4153-ae93-9a890d068c79

**Comment by JamesTKhan** [comment]
Closing out since we have PR's in progress on this.
